### DISCUSSION TITLE
Using getStaticProps with Docker

### DISCUSSION THREAD

**Discussion by 2stash** [vercel/next.js#35149] [Help]
Hello. How can you deploy nextjs with docker. I am deploying a nextjs frontend with node.js backend and I can't figure it out. The main issue is that nextjs tries to run getStaticProps and getServerSideProps when building the docker image. I would have to have the backend running, and have it open to receive requests from a docker build command.  In a react app for example, if I had ran docker-compose build, it would build, and then I would then just have to make sure the backend is running when the front end is in use. So "docker-compose up -d backend", "docker-compose up -d frontend".  I can't currently get this to work with getStaticProps because the backend is not available when building the frontend docker image.

The cicd pipeline I currently use is I push to gitlab, and that builds the dockerfile and automatically pushes to openshift. I don't believe that when the docker build command is run on gitlab it has access to anything, so even if the backend api is running it can't access it.

I've read through the docs:
https://github.com/vercel/next.js/blob/canary/docs/deployment.md

and a similar discussion: https://github.com/vercel/next.js/discussions/16995

**Comment by taco-chen** [comment]
Just met the same issue. Looking forward to some solutions here.

**Reply by mahershala2k18** [reply]
me too
